### PR TITLE
Allow converter failures to fail workflow and other minor things

### LINKED ISSUE #238: [Bug] Can't debugging workflow using vscode
when run a workflow with vscode debugger attach I get the flowwing error as soon a workflow start.

```File "/usr/local/lib/python3.7/site-packages/debugpy/_vendored/pydevd/pydevd.py", line 1715, in process_internal_commands
    int_cmd.do_it(self)
  File "/usr/local/lib/python3.7/site-packages/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_comm.py", line 542, in do_it
    self.method(dbg, *self.args, **self.kwargs)
  File "/usr/local/lib/python3.7/site-packages/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_constants.py", line 511, in new_func
    with warnings.catch_warnings():
  File "/usr/local/lib/python3.7/warnings.py", line 458, in __init__
    self._module = sys.modules['warnings'] if module is None else module
  File "/usr/local/lib/python3.7/site-packages/temporalio/worker/workflow_sandbox/_importer.py", line 393, in __getitem__
    return self.current[key]
KeyError: 'warnings'
```

### Environment/Versions

- OS and processor: [Linux]
- Temporal Version: Python sdk 0.1b4
- Are you using Docker or Kubernetes or building Temporal from source? Nor

### LINKED ISSUE #326: [Feature Request] Add helper for checking whether exception represents cancellation
### Describe the solution you'd like

Logic is something like:

```python
is_cancelled =
    isinstance(err, asyncio.CancelledError) or
    isinstance(err, temporalio.exceptions.CancelledError) or
    (
        (isinstance(err, temporalio.exceptions.ActivityError) or isinstance(err, temporalio.exceptions.ChildWorkflowError)) and
        isinstance(err.cause, temporalio.exceptions.CancelledError)
    )
```

The internal workflow code can also use this

### LINKED ISSUE #327: Use stable pydoctor version instead of devel version
Pydoctor's version in use is a custom development version. Since support for `@overload` has now been released, the stable version should be used instead.

### LINKED ISSUE #328: Remove "similar to Pedantic" from no-generic part of README
They do have limited support now

### PR DESCRIPTION
**Pull request by cretz**
## What was changed

* Allow converter failures to fail workflow. We intentionally did not do codec at this time because it is more complicated.
* Remove build info dump in CI
* Remove Pydantic README reference
* Update pydoctor
* Add `temporalio.exceptions.is_cancelled_exception` helper function

## Checklist

1. Relates to #321
2. Closes #328
3. Closes #327
3. Closes #326

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -306,8 +306,8 @@ This notably doesn't include any `date`, `time`, or `datetime` objects as they m
 Users are strongly encouraged to use a single `dataclass` for parameter and return types so fields with defaults can be
 easily added without breaking compatibility.
 
-Classes with generics may not have the generics properly resolved. The current implementation, similar to Pydantic, does
-not have generic type resolution. Users should use concrete types.
+Classes with generics may not have the generics properly resolved. The current implementation, does not have generic
```
Comment (Sushisource) [inline README.md:309]:
```suggestion
Classes with generics may not have the generics properly resolved. The current implementation does not have generic
```

File: temporalio/exceptions.py [hunk 2]
```diff
@@ -322,3 +323,32 @@ def started_event_id(self) -> int:
...
+    those latter two have a :py:class:`CancelledError` cause.
+
+    Args:
+        exception: Exception to check.
+
+    Returns:
+        True if a cancelled exception, false if not.
+    """
+    return (
+        isinstance(exception, asyncio.CancelledError)
```
Comment (Sushisource) [inline temporalio/exceptions.py:345]: Oh boy this whole thing is not fun lol

File: temporalio/worker/_workflow_instance.py [hunk 3]
```diff
@@ -287,7 +288,17 @@ def activate(
                 # be checked in patch jobs (first index) or query jobs (last
                 # index).
                 self._run_once(check_conditions=index == 1 or index == 2)
+        except temporalio.exceptions.FailureError as err:
+            # We want failure errors during activation, like those that can
+            # happen during payload conversion, to fail the workflow not the
+            # task
+            try:
+                self._set_workflow_failure(err)
```
Comment (Sushisource) [inline temporalio/worker/_workflow_instance.py:296]: Making sure - this is only happening if they explicitly raise something like `ApplicationError` - not on any old random exception, correct?

**Review by Sushisource — APPROVED** [review]
LGTM, just one question confirming the behavior

File: temporalio/exceptions.py (hunk 2 above)
Comment (cretz) [reply temporalio/exceptions.py:345]: Yeah, we basically write this same thing in every SDK

File: temporalio/worker/_workflow_instance.py (hunk 3 above)
Comment (cretz) [reply temporalio/worker/_workflow_instance.py:296]: Correct, this catch clause is for `temporalio.exceptions.FailureError` only

**Pull request merged**